genai_factory/src/genai_factory/__main__.py

```diff
@@ -13,100 +13,93 @@
+    config_path: pathlib.Path,
     runner: str,
-    path: str,
-    workflow_type: str,
 ):
     """
-    Run workflow application
+    Run given workflows from file.
 
-    :param workflow_name:   The workflow name
-    :param runner:          The runner to use, default is fastapi.
-    :param path:            The path to the workflow file.
-    :param workflow_type:   The type of the workflow. Can be one of mlrun.genai.schemas.WorkflowType
+    :param workflows_path: The workflows file path.
+    :param config_path:    A configuration file path. Note: This option takes priority on both set configuration via
+                           the workflow server instance in code, and via path from environment variable:
 
-    :return:    None
+                           1. Command config
+                           2. Env file config
+                           3. Workflow server manual set config
+    :param runner:         The runner to use, default is fastapi.
     """
-    # Import the workflow's graph from the path
-    if path:
-        # Load the module from the given file path
-        spec = importlib.util.spec_from_file_location("module_name", path)
-        module = importlib.util.module_from_spec(spec)
-        sys.modules["module_name"] = module
-        spec.loader.exec_module(module)
-
-        # Retrieve the desired object from the module
-        click.echo(f"Using graph from {path}")
-        graph = getattr(module, "workflow_graph")
-    else:
-        # Use the default graph
-        click.echo("Using default graph")
-        graph = default_graph
-
-    if runner == "nuclio":
-        click.echo("Running nuclio is not supported yet")
-    elif runner == "fastapi":
-        app_server = AppServer()
-        app_server.add_workflow(
-            project_name="default",
-            name=workflow_name,
-            graph=graph,
-            deployment=config.infer_path(workflow_name),
-            workflow_type=workflow_type,
-            username=username,
-            update=True,
-        )
-        app = app_server.to_fastapi(router=router)
-
-        # Deploy the fastapi app
-        host = config.workflow_deployment["host"]
-        port = config.workflow_deployment["port"]
-        click.echo(f"Running workflow {workflow_name} with fastapi on {host}")
-        uvicorn.run(app, host=host, port=port)
-
+    # Load the configuration if given:
+    if config_path:
+        click.echo(f"Loading configuration from: {config_path}")
+        config = WorkflowServerConfig.from_yaml(config_path)
     else:
-        click.echo(
-            f"Runner {runner} not supported. Supported runners are: nuclio, fastapi"
-        )
+        # Initialize the global configuration:
+        if _CONFIG_PATH:
+            config = WorkflowServerConfig.from_yaml(_CONFIG_PATH)
+        elif _IS_LOCAL_CONFIG:
+            config = WorkflowServerConfig.local_config()
+        else:
+            config = WorkflowServerConfig()
```

Comment:
We don't want to run over user set configuration...

```suggestion
            config = None  # The user count on the default config or he set it manually in the script.
```